### ISSUE TITLE
Captcha Audio Solver

### ISSUE THREAD

**Issue by gmcsnetwork** [mdesilva/AmazonFlexUnlimited#103]
Anyone else looking for ways to programmatically complete the new Offer verification? So far, I have been using the app to complete the captcha and continue to run the script. 

I found the URL to get to the captcha page, and while inspecting it noticed it uses arkoselabs.com captcha solution. That lead me to another GitHub repository with a script that could potentially solve the captcha using the audio option. 

https://www.amazon.com/aaut/verify/flex-offers?options=%7B%22clientData%22%3Anull%2C%22challengeType%22%3A%22ARKOSE_LEVEL_1%22%2C%22locale%22%3A%22en-US%22%2C%22enableHeaderFooter%22%3Atrue%2C%22enableBypassMechanism%22%3Afalse%2C%22enableModalView%22%3Afalse%2C%22fwcimBlob%22%3Anull%7D

https://github.com/useragents/Funcaptcha-Audio-Solver

Wondering what others here are doing to get around the new verification.

**Comment by possiblemeatball** [comment]
Scratch that previous comment, audio challenges are included

**Comment by possiblemeatball** [comment]
In case of the linked repository, the public_key would have to be extracted from the captcha page, and even given that it appears that others have had issue with Amazon captchas [(seen in this issue here)](https://github.com/useragents/Funcaptcha-Audio-Solver/issues/9)

**Comment by ransel2016** [comment]
I know of more than two scripts that have already implemented the resolution of the captcha automatically and working well, hopefully and here we find a solution. Greetings

**Comment by travislazzaro** [comment]
> Anyone else looking for ways to programmatically complete the new Offer verification? So far, I have been using the app to complete the captcha and continue to run the script.
> 
> I found the URL to get to the captcha page, and while inspecting it noticed it uses arkoselabs.com captcha solution. That lead me to another GitHub repository with a script that could potentially solve the captcha using the audio option.
> 
> https://www.amazon.com/aaut/verify/flex-offers?options=%7B%22clientData%22%3Anull%2C%22challengeType%22%3A%22ARKOSE_LEVEL_1%22%2C%22locale%22%3A%22en-US%22%2C%22enableHeaderFooter%22%3Atrue%2C%22enableBypassMechanism%22%3Afalse%2C%22enableModalView%22%3Afalse%2C%22fwcimBlob%22%3Anull%7D
> 
> https://github.com/useragents/Funcaptcha-Audio-Solver
> 
> Wondering what others here are doing to get around the new verification.

hey, have you implemented that captcha script?

**Comment by gmcsnetwork** [comment]
No, I'm still manually completing the verification in the app and restarting the script.

**Comment by travislazzaro** [comment]
yeah same. it's not that big of a deal but kind of annoying to try and manually get a block to solve the captcha only to not even get the block after solving it lol

**Comment by jaimbox** [comment]
Hello everyone, I have tried this solution and it works for me. It is a paid solution but it solves the problem.

1. You must create an account [here](http://getcaptchasolution.com/hsrx4hiswv). 
2. Install the library `pip install python3-anticaptcha`
3. Import the library `from anticaptchaofficial.funcaptchaproxyless import *`
4. When you get the 307 error paste this:
 ```
solver = funcaptchaProxyless()
solver.set_verbose(1)
solver.set_key("YOUR_KEY")
solver.set_website_url("https://www.amazon.com/aaut/verify/flex-offers/challenge?challengeType=ARKOSE_LEVEL_1&returnTo=https://www.amazon.com&headerFooter=false")
solver.set_website_key("2F1CD804-FE45-F12B-9723-240962EBA6F8")

token = solver.solve_and_return_solution()
if token != 0:
	print("result token: " + token)
else:
	print("task finished with error " + solver.error_code)
```

**Comment by Vasa211** [comment]
@jaimbox your code working now? Because I all time receiving ERROR_CAPTCHA_UNSOLVABLE

**Comment by BimBimSalaBim** [comment]
> 1. You must create an account [here](http://getcaptchasolution.com/hsrx4hiswv).

just a heads up this is a referral link. think that should have been mentioned...

**Comment by vtdeleond** [comment]
¬_¬
